feat(cli): add delete-users-by-role command

kariae opened the pull request:
# Add delete-users-by-role command

This PR introduces a new WP-CLI command `delete-users-by-role` that allows for bulk deletion of WordPress users based on their roles.

## Features
- Delete users by specifying roles to delete:
  ```bash
  wp newspack-migration-tools delete-users-by-role --roles=subscriber,author
  ```
- Delete users by excluding specific roles:
  ```bash
  wp newspack-migration-tools delete-users-by-role --exclude-roles=administrator,editor
  ```
- Optional post reassignment:
  ```bash
  wp newspack-migration-tools delete-users-by-role --roles=subscriber --reassign-to=1
  ```
- Dry-run mode for safety:
  ```bash
  wp newspack-migration-tools delete-users-by-role --roles=subscriber --dry-run
  ```

## Implementation Details
- Validates roles before processing
- Handles post reassignment/deletion
- Includes detailed logging
- Includes comprehensive test suite
- Includes workaround for [Co-Authors Plus issue #1096](https://github.com/Automattic/Co-Authors-Plus/issues/1096)

## Testing Instructions
1. Install and activate the plugin
2. Run the command with `--dry-run` to preview changes:
   ```bash
   wp newspack-migration-tools delete-users-by-role --roles=subscriber --dry-run
   ```
3. Run without `--dry-run` to perform actual deletion
4. Test with both `--roles` and `--exclude-roles` parameters
5. Test with and without `--reassign-to` parameter

## QA Checklist
- [x] Code follows WordPress coding standards
- [x] All tests pass
- [x] Documentation is clear and complete
- [x] Command handles edge cases gracefully

kariae committed:
feat(cli): add delete-users-by-role command

This command allows bulk deletion of WordPress users based on their roles. Users can be deleted either by specifying roles to delete or roles to preserve.

Features:\n- Delete users by role(s)\n- Exclude specific roles from deletion\n- Optional post reassignment\n- Dry-run mode for safety\n- Detailed logging\n\nIncludes workaround for Co-Authors Plus issue #1096

naxoc commented on tests/Command/TestUsersMigrator.php:
```diff
@@ -0,0 +1,452 @@
+	/**
+	 * Clean up after all tests.
+	 */
+	public function tearDown(): void {
```
Does all this (maybe except for the wp hook/removal) not happen automatically? I've never had to tear down stuff like this, but I could be wrong.

kariae replied:
Good point. I deleted the tearDown method.

jorostoyanov reviewed (approved):
Awesome! Works exactly as described 🙌

jorostoyanov reviewed (changes requested):
Hey @kariae  👋 

Do you think it's worth adding a confirm step if the `--reassign-to` option is not passed? Otherwise, we can accidentally delete Posts this way.

I just had this happened to me and wanted to make sure we prevent this from happening again 🙌

kariae committed:
fix(UsersMigrator): force reassigning posts to a user when deleting users

kariae commented:
Hey @jorostoyanov 👋 

[531c355](https://github.com/Automattic/newspack-migration-tools/pull/57/commits/531c35592c3aef6f2913f746f47164d2890e0f7a) forces setting the `reassign-to` parameter so we can avoid accidentally deleting posts.

kariae committed:
Merge branch 'trunk' into feat/delete-users-by-role

kariae committed:
test(UsersMigrator): make reassign-to parameter mandatory in delete-users-by-role

- Remove test_delete_users_without_reassignment test since it's no longer valid
- Update test_delete_users_with_exclude_roles to verify posts are reassigned instead of deleted
- Add reassign-to parameter to all test cases that were missing it

This change ensures that posts are always reassigned to a valid user when deleting users,
preventing orphaned posts in the database.

kariae committed:
refactor(TestUsersMigrator): remove tearDown method

kariae commented:
@jorostoyanov @naxoc can I get a final check on this, please?

naxoc reviewed (approved):
This looks good! A small thing: no need to cleanup users in the test I think. Each function is run on a clean DB.
